In [1]:
import pandas as pd
import sqlite3 as db

## Part A

In [2]:
# Establish connection
connection = db.connect("dj.db")

# Define cursor to submit requests and hold results
cursor = connection.cursor()

#Run SQL Statement
cursor.execute('SELECT id, ticker, name from tickers')

# Get result of SQL statement request
rows = cursor.fetchall()

# Implement pandas
df = pd.DataFrame(rows)

#Establish column names
df.columns =['id','ticker','name']

#Set index
df.set_index('ticker', inplace = True)

df

id                                         name
ticker                                                 
M        1                                   Macy's Inc
AXP      2                     American Express Company
AAPL     3                                   Apple Inc.
BA       4                           The Boeing Company
CAT      5                             Caterpillar Inc.
CVX      6                          Chevron Corporation
CSCO     7                           Cisco Systems Inc.
KO       8                      Coca-Cola Company (The)
DIS      9                      The Walt Disney Company
DOW     10                                     Dow Inc.
XOM     11                      Exxon Mobil Corporation
GS      12               Goldman Sachs Group Inc. (The)
HD      13                        Home Depot Inc. (The)
IBM     14  International Business Machines Corporation
INTC    15                            Intel Corporation
JNJ     16                            Johnson & Johnson
JPM     17                        JP Morgan Chase & Co.
MCD     18                       McDonald's Corporation
MRK     19                   Merck & Company Inc. (new)
MSFT    20                        Microsoft Corporation
NKE     21                                    Nike Inc.
PFE     22                                  Pfizer Inc.
PG      23               Procter & Gamble Company (The)
TRV     24                 The Travelers Companies Inc.
UTX     25              United Technologies Corporation
UNH     26         UnitedHealth Group Incorporated (DE)
VZ      27                  Verizon Communications Inc.
V       28                                    Visa Inc.
WMT     29                                 Walmart Inc.
WBA     30                Walgreens Boots Alliance Inc.

## Part B

In [3]:
#Write SQL query
query = """
SELECT min(date) as first,max(date) as last FROM daily_measures
"""
#Reveal SQL query
pd.read_sql_query(query,connection)

first        last
0  2017-01-03  2017-12-29

## Part C

In [4]:
#Write SQL query
query = """
SELECT date,max(closing) from daily_measures where ticker_id = 3 ORDER BY date
"""
#Run SQL Statement
cursor.execute(query)
#Display SQL Output
for row in cursor:
   print(row)

('2017-12-18', 176.42)


#### Highest price for Apple is $176.42 on December 18, 2017

## Part D

In [5]:
#Write Query
query = """
SELECT a.ticker,a.name,b.opening,b.closing,PRINTF('%1.2f%%',(b.closing-b.opening)/b.opening*100) AS change 
FROM tickers a left join daily_measures b on a.id = b.ticker_id WHERE ticker = 'AAPL'
"""

#Reveal Query Result
pd.read_sql_query(query,connection)


ticker        name  opening     closing  change
0     AAPL  Apple Inc.   115.80  114.715378  -0.94%
1     AAPL  Apple Inc.   115.85  114.586983  -1.09%
2     AAPL  Apple Inc.   115.92  115.169696  -0.65%
3     AAPL  Apple Inc.   116.78  116.453639  -0.28%
4     AAPL  Apple Inc.   117.95  117.520300  -0.36%
..     ...         ...      ...         ...     ...
244   AAPL  Apple Inc.   175.03  174.540000  -0.28%
245   AAPL  Apple Inc.   175.05  174.090000  -0.55%
246   AAPL  Apple Inc.   175.10  174.970000  -0.07%
247   AAPL  Apple Inc.   175.11  174.670000  -0.25%
248   AAPL  Apple Inc.   175.11  175.880000   0.44%

[249 rows x 5 columns]